java-symbol-solver-core/src/main/java/me/tomassetti/symbolsolver/javaparsermodel/declarations/JavaParserSymbolDeclaration.java

```diff
@@ -129,7 +129,11 @@ public Type getType() {
             VariableDeclarator variableDeclarator = (VariableDeclarator) wrappedNode;
             if (getParentNode(wrappedNode) instanceof VariableDeclarationExpr) {
                 VariableDeclarationExpr variableDeclarationExpr = (VariableDeclarationExpr) getParentNode(variableDeclarator);
-                return JavaParserFacade.get(typeSolver).convert(variableDeclarationExpr.getElementType(), JavaParserFactory.getContext(wrappedNode, typeSolver));
+                Type type = JavaParserFacade.get(typeSolver).convert(variableDeclarationExpr.getElementType(), JavaParserFactory.getContext(wrappedNode, typeSolver));
+                for (int i=0;i<variableDeclarationExpr.getArrayBracketPairsAfterElementType().size();i++) {
+                    type = new ArrayType(type);
+                }
```

Comment:
See last remark! And if you really must do wrapping, check the static methods on ArrayType :)